packages/cli/lib/DevServerManager.js

```diff
@@ -33,75 +40,99 @@ class DevServerManager {
 
     for (let i = 0; i < serverKeys.length; i++) {
       const serverKey = serverKeys[i];
-      const serverInterface = this.devServers[serverKey];
-      await callback(serverInterface, serverKey);
+      const devServer = this.devServers[serverKey];
+
+      const compatibleComponents =
+        this.componentsByType[devServer.componentType] || {};
+
+      if (Object.keys(compatibleComponents).length) {
```

Comment:
I am now only calling the lifecycle methods for the dev server interfaces if there are available components for the server to interact with.